CONTRIBUTING.md

```diff
@@ -156,6 +156,31 @@ To format and lint the codes,
 tox -e fmt,lint
 ```
 
+## [Optional] Documentation setup
+
+This repository uses [mkdocs](https://github.com/mkdocs/mkdocs/tree/master) to generate a documentation from python docstring. As long as you write docstrings for your code, the document (API references section) will be automatically updated when the code is merged into `main` branch, so there is no need for you to do anything.
+
+If you want to manually edit the document, please run the following commands to setup the editing environment.
+
+```bash
```

Comment:
If we do go the route or scripting and tracking deps in `pyproject` this will need to be updated

Reply:
I don't think we have to chose this route. Please check  https://github.com/IBM/oper8/pull/147#discussion_r1894586705